data_science/scripts/asg_utils.py

```diff
@@ -17,34 +12,34 @@ def discover_data_science_asg(asg_client):
     # to check!
     resp = asg_client.describe_auto_scaling_groups(MaxRecords=100)
 
-    for asg_data in resp['AutoScalingGroups']:
 
-        # The structure of the response is a little awkward.  It's a list of
-        # entries of the form:
-        #
-        #   {'Key': '<KEY>',
-        #    'PropagateAtLaunch': (True|False),
-        #    'ResourceId': '<RESOURCE_ID>',
-        #    'ResourceType': 'auto-scaling-group',
-        #    'Value': '<VALUE>'}]
-        #
-        # We only care about the tag values, so we extract them into a
-        # Python dict.
+    # The structure of the response is a little awkward.  It's a list of
+    # entries of the form:
+    #
+    #   {'Key': '<KEY>',
+    #    'PropagateAtLaunch': (True|False),
+    #    'ResourceId': '<RESOURCE_ID>',
+    #    'ResourceType': 'auto-scaling-group',
+    #    'Value': '<VALUE>'}]
+    #
+    # We only care about the tag values, so we extract them into a
+    # Python dict.
+    for asg_data in resp['AutoScalingGroups']:
         actual_tags = {t['Key']: t['Value'] for t in asg_data['Tags']}
 
-        if actual_tags == DATA_SCIENCE_ASG_TAGS:
+        if ('name' in actual_tags) and (actual_tags['name'] == tag_name):
```

Comment:
Neater: assuming `name` will be not-None (add an assert for this if you want), you can do:

```python
if actual_tags.get(name) == tag_name:
```

which gets the value from the dict if present, or None if not.

Reply:
I did wonder if there were a safe way to do this 👍